src/Config/Parser/AnnotationParser.php

```diff
@@ -253,19 +253,20 @@ private static function typeAnnotationToGQLConfiguration(
     ): array {
         $isMutation = $isDefault = $isRoot = false;
         if (isset($configs['definitions']['schema'])) {
+            $defaultSchemaName = isset($configs['definitions']['schema']['default']) ? 'default' : array_key_first($configs['definitions']['schema']);
             foreach ($configs['definitions']['schema'] as $schemaName => $schema) {
                 $schemaQuery = $schema['query'] ?? null;
                 $schemaMutation = $schema['mutation'] ?? null;
 
                 if ($schemaQuery && $gqlName === $schemaQuery) {
                     $isRoot = true;
-                    if ('default' == $schemaName) {
+                    if ($defaultSchemaName == $schemaName) {
```

Comment:
can we use strict check here?

Reply:
And a think the "if" conditions could also be simplified with `$schemaQuery === $gqlName` and `$schemaMutation === $gqlName`